src/binary/non_blocking/raw_binary_reader.rs

```diff
@@ -1361,6 +1362,9 @@ impl<'a, A: AsRef<[u8]>> TxReader<'a, A> {
                 return Ok(Some(RawStreamItem::Nothing));
             }
             *type_descriptor = self.tx_buffer.peek_type_descriptor()?;
+            if type_descriptor.is_ivm_start() {
+                return self.read_ivm().map(Some);
```

Comment:
I assume IVM at non-top-level is caught elsewhere?

Reply:
Yeah, `read_ivm()` returns an error if the reader isn't at the top level.